Tests/helper.py

```diff
@@ -165,6 +168,18 @@ def assert_tuple_approx_equal(
             pytest.fail(msg + ": " + repr(actuals) + " != " + repr(targets))
 
 
+def running_in_another_thread() -> bool:
+    frameInfo = inspect.stack()[1]
+    identifier = (
+        frameInfo.filename + ":" + frameInfo.function + ":" + str(frameInfo.lineno)
+    )
+    if identifier in already_running:
+        return True
+
+    already_running.append(identifier)
```

Comment:
I think that this introduces another possible race, where two or more threads all think that the test is not currently running elsewhere and add themselves to the list.